java/extentreports/src/main/java/com/relevantcodes/extentreports/Report.java

```diff
@@ -228,7 +228,7 @@
     
     protected SuiteTimeInfo suiteTimeInfo;
     protected SystemInfo systemInfo;
-    protected List<ExtentTest> testList;
+    protected List<ExtentTest> testList = new ArrayList<ExtentTest>();
```

Comment:
This and the ExtentReports.java fix is #320 , Couldn't figure out if its the same null pointer or not. But if you think this is a new issue change the merge description.

I am not sure about the business requirements. If no tests are added do we really want to generate an empty report or not. In this fix I have just fixed the exception, business rules needs to be implemented separately.